### PR TITLE
batches: remove batch specs created with CLI from site admin list

### LINKED ISSUE #36020: Batch spec list shows wrong data for src-cli workflow
The list of batch specs in the site admin area shows incorrect execution times for src-cli. We should hide it from the UI and check if we can not return it in the API.

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes #36020 

## Test plan

* Create a couple of batch changes via `src-cli`
* Validate they don't show up in the Batch Specs page.

BEFORE:
CleanShot 2022-06-06 at 12 15 34@2x

AFTER:
CleanShot 2022-06-06 at 12 13 05@2x

### CODE REVIEW

**Review by Piszmog — APPROVED** [review]

**Comment by BolajiOlajide** [comment]
> A couple things:
> 
> * I think I would make `ExcludeNonSSBCSpecs = true` the default -- listing batch specs wasn't a meaningful operation until SSBC anyway.
> * I'm wondering about the flag name. Is it easier if we're describing it from an `includes` perspective, like `IncludeLocallyExecutedSpecs`?
> * Let's make sure to send this flag with the list of batch specs we're querying from the batch change details tab [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/enterprise/batches/backend.ts?L36%3A14=), too!

* I have no problem with `IncludeLocallyExecutedSpecs` - I'll make the update.
Thanks @courier-new

File: cmd/frontend/graphqlbackend/batches.graphql [hunk 1]
```diff
@@ -1842,6 +1842,10 @@ extend type Query {
         Opaque pagination cursor.
         """
         after: String
+        """
+        Include locally executed batch specs in the list of specs returned.
```
Comment (courier-new) [inline cmd/frontend/graphqlbackend/batches.graphql:1846]: Maybe just to make it EXTRA clear:

```suggestion
        Include locally executed (i.e. with src-cli) batch specs in the list of specs returned.
```

File: cmd/frontend/graphqlbackend/batches.graphql [hunk 2]
```diff
@@ -2699,6 +2703,10 @@ type BatchChange implements Node {
...
```
Comment (courier-new) [inline cmd/frontend/graphqlbackend/batches.graphql:2707]:
```suggestion
        Include locally executed (i.e. with src-cli) batch specs in the list of specs returned.
```

File: enterprise/internal/batches/store/batch_specs_test.go [hunk 3]
```diff
@@ -20,10 +20,12 @@ import (
 )
 
 func testStoreBatchSpecs(t *testing.T, ctx context.Context, s *Store, clock ct.Clock) {
-	batchSpecs := make([]*btypes.BatchSpec, 0, 3)
+	batchSpecs := make([]*btypes.BatchSpec, 0, 4)
 
 	t.Run("Create", func(t *testing.T) {
 		for i := 0; i < cap(batchSpecs); i++ {
+			// only the fourth batch spec should have be locally-created
```
Comment (courier-new) [inline enterprise/internal/batches/store/batch_specs_test.go:27]:
```suggestion
			// only the fourth batch spec should be locally-created
```

File: client/web/src/enterprise/batches/BatchSpecsPage.tsx [hunk 4]
```diff
@@ -69,6 +71,7 @@ export const BatchChangeBatchSpecList: React.FunctionComponent<
 export interface BatchSpecListProps extends ThemeProps, Pick<RouteComponentProps, 'history' | 'location'> {
     currentSpecID?: Scalars['ID']
     queryBatchSpecs?: typeof _queryBatchSpecs
+    includeLocallyExecutedSpecs: Scalars['Boolean']
```
Comment (courier-new) [inline client/web/src/enterprise/batches/BatchSpecsPage.tsx:74]: We only ever pass this as `false` in the places we use this component, right? I'd be tempted to drop it as a prop and just hardcode it in the `passedArguments` instead.

**Review by courier-new — APPROVED** [review]
Awesome, thanks for the updates Bolaji!! Looks great to me, just left a couple really minor comments.

File: client/web/src/enterprise/batches/BatchSpecsPage.tsx (hunk 4 above)
Comment (BolajiOlajide) [reply client/web/src/enterprise/batches/BatchSpecsPage.tsx:74]: Okay, I'll make the change.

**Pull request merged**